qt/scientific_interfaces/Muon/ALCDataLoadingModel.cpp

```diff
@@ -0,0 +1,346 @@
+#include <algorithm>
+
+namespace {
+bool is_decimal(const char character) { return character == '.'; }
+} // namespace
+
+using namespace Mantid::API;
+
+namespace MantidQt::CustomInterfaces {
+
+ALCDataLoadingModel::ALCDataLoadingModel()
+    : m_numDetectors(0), m_loadingData(false), m_directoryChanged(false), m_lastRunLoadedAuto(-2),
+      m_wasLastAutoRange(false), m_minTime(0) {}
+
+void ALCDataLoadingModel::setLoadingData(bool isLoading) { m_loadingData = isLoading; }
+bool ALCDataLoadingModel::getLoadingData() { return m_loadingData; }
+
+void ALCDataLoadingModel::setLoadedData(const MatrixWorkspace_sptr &data) { m_loadedData = data; }
+MatrixWorkspace_sptr ALCDataLoadingModel::getLoadedData() { return m_loadedData; }
+
+MatrixWorkspace_sptr ALCDataLoadingModel::exportWorkspace() {
+  if (m_loadedData)
+    return std::const_pointer_cast<MatrixWorkspace>(m_loadedData);
+  return MatrixWorkspace_sptr{};
+}
+
+/**
+ * Load new data and update the view accordingly
+ * @param files :: [input] range of files (user-specified or auto generated)
+ */
+void ALCDataLoadingModel::load(const IALCDataLoadingView *view) {
+
+  IAlgorithm_sptr alg = AlgorithmManager::Instance().create("PlotAsymmetryByLogValue");
+  alg->setAlwaysStoreInADS(false); // Don't want workspaces in the ADS
+
+  // Change first last run to WorkspaceNames
+  alg->setProperty("WorkspaceNames", m_filesToLoad);
+  alg->setProperty("LogValue", view->log());
+  alg->setProperty("Function", view->function());
+  alg->setProperty("Type", view->calculationType());
+  alg->setProperty("DeadTimeCorrType", view->deadTimeType());
+  alg->setProperty("Red", view->redPeriod());
+
+  // If time limiting requested, set min/max times
+  if (auto timeRange = view->timeRange()) {
+    double timeMin = (*timeRange).first;
+    double timeMax = (*timeRange).second;
+    if (timeMin >= timeMax) {
+      throw std::invalid_argument("Invalid time limits");
+    }
+    alg->setProperty("TimeMin", timeMin);
+    alg->setProperty("TimeMax", timeMax);
+  }
+
+  // If corrections from custom file requested, set file property
+  if (view->deadTimeType() == "FromSpecifiedFile") {
+    alg->setProperty("DeadTimeCorrFile", view->deadTimeFile());
+  }
+
+  // If custom grouping requested, set forward/backward groupings
+  if (view->detectorGroupingType() == "Custom") {
+    alg->setProperty("ForwardSpectra", view->getForwardGrouping());
+    alg->setProperty("BackwardSpectra", view->getBackwardGrouping());
+  }
+
+  // Set alpha for balance parameter
+  alg->setProperty("Alpha", view->getAlphaValue());
+
+  // If Subtract checkbox is selected, set green period
+  if (view->subtractIsChecked()) {
+    alg->setProperty("Green", view->greenPeriod());
+  }
+
+  alg->setPropertyValue("OutputWorkspace", "__NotUsed");
+
+  // Execute async so we can show progress bar
+  Poco::ActiveResult<bool> result(alg->executeAsync());
+  while (!result.available()) {
+    QCoreApplication::processEvents();
+  }
```

Comment:
**No need to make any changes with regards to this comment**

This is something to be refactored/removed in a future PR. I'm sorry you had to come across this! It is trying to keep the GUI responsive while the algorithm is executed in a background thread, and whilst waiting for the algorithm to finish. It is much preferred to have a "callback" function that gets called when an algorithm is completed. This is functionality provided by the `AlgorithmRunner` class. This change can be complicated so best not to make any changes as part of this PR

Reply:
Sounds good!